source/common/grpc/google_grpc_utils.cc

```diff
@@ -78,7 +78,7 @@ Buffer::InstancePtr GoogleGrpcUtils::makeBufferInstance(const grpc::ByteBuffer&
   // NB: ByteBuffer::Dump moves the data out of the ByteBuffer so we need to ensure that the
   // lifetime of the Slice(s) exceeds our Buffer::Instance.
   std::vector<grpc::Slice> slices;
-  byte_buffer.Dump(&slices);
+  RELEASE_ASSERT(byte_buffer.Dump(&slices) == grpc::Status::OK, "");
```

Comment:
Please explain what happens in the rc details failure.  Should this only happen if the machine is out of memory?  If it can otherwise fail we should probably do error handling instead

Reply:
@alyssawilk Good call!

It looks like this can fail for various reasons (e.g. uninitialized buffer, decompression errors when input buffer is compressed), but `GoogleGrpcUtils::makeBufferInstance()` is only used in tests, so I'm not sure how much we care about those...

@jplevyak any plans for this function to be used for real traffic? Perhaps we should move it into tests?

Reply:
https://github.com/envoyproxy/envoy/pull/6525 is under active review

Reply:
If it's only in test today I'd also be OK with landing this as-is to unblock merge with a TODO to do the error handling as part of #6525, if @jplevyak is up for doing the error handling.

Reply:
Oops, sorry, I totally missed this one.

@jplevyak do you want to incorporate those changes into #6525 or should I return `nullptr` in case of errors in this PR?

Reply:
I can do the error handling.  The question is what is most appropriate. Since the ByteBuffer is coming from the low level gRPC library, the most likely causes would be a bug in that library or memory corruption.  This would suggest that a RELEASE_ASSERT at that level would be appropriate.  WDYT?  Alternatively an Exception or at critical log?

Reply:
From looking at the gRPC code, it seems that `Dump()` (well, `grpc_byte_buffer_reader_init()`) [can fail because of decompression errors](https://github.com/grpc/grpc/blob/a3366c1b7f53e825e6380c5fb8cd096172fe7b00/src/core/lib/surface/byte_buffer_reader.cc#L61), which means that if someone sends corrupted gRPC message, then `Dump()` would return error, so we should treat it as invalid request/response, and not runtime error.